File: Python/alphabet/alphabet.py


arr = input()


ls = [1]*(len(arr))

for i in reversed(range(len(arr))):
    for j in range(i, len(arr)):
        if arr[i] < arr[j]:
            ls[i] = max(ls[i], ls[j]+1)


print(26-max(ls))

File: alphabet/Alphabet.java
package alphabet;

import java.util.*;

public class Alphabet {


    public static void main(String[] args) {



        Scanner scan = new Scanner(System.in);

        char[] input = scan.nextLine().toCharArray();
        scan.close();
        
        int[] arr = new int[input.length];
        for (int i : arr) {
            i = 1;
        }

        for (int i = input.length-1; i >= 0; i--) {

            char a = input[i];

            for (int j = i; j < input.length; j++) {
                char b = input[j];

                if(a<b && arr[j]+1>arr[i]) {
                    arr[i] = arr[j]+1;
                }
            }
        }

        int max = 0;
        for (int i : arr) {
            if(i > max) {
                max = i;
            }
        }

        System.out.println(26-max);

    }


}